File: 백준/Silver/10845. 큐/큐.py
import sys
N = int(input())
stack = []
for i in range(N):
  cmd = sys.stdin.readline().split()
  if cmd[0] == 'push':
    stack.append(cmd[1])
  elif cmd[0] == 'pop':
    if len(stack) != 0:
      print(stack.pop(0))
    else:
      print(-1)
  elif cmd[0] == 'size':
    print(len(stack))
  elif cmd[0] == 'empty':
    if len(stack) == 0:
      print(1)
    else:
      print(0)
  elif cmd[0] == 'front':
    if len(stack) != 0:
      print(stack[0])
    else:
      print(-1)
  elif cmd[0] == 'back':
    if len(stack) != 0:
      print(stack[-1])
    else:
      print(-1)

File: 백준/Silver/10845. 큐/큐.cc
#include <iostream>
#include <bits/stdc++.h>

using namespace std;

int main() {
	ios::sync_with_stdio(0);
	cin.tie(0);

	queue<int> Q;
	int n;
	cin >> n;

	while(n--){
		string cmd;
		cin >> cmd;
		if(cmd == "push"){
			int value;
			cin >> value;

			Q.push(value);
		}
		if(cmd == "pop"){
			if(Q.empty()) cout << -1 << "\n";
			else{
				cout << Q.front() << "\n";
				Q.pop();
			}
		}
		if(cmd == "size") cout << Q.size() << "\n";
		if(cmd == "empty"){
			if(Q.empty()) cout << 1 << "\n";
			else cout << 0 << "\n";
		}
		if(cmd == "front"){
			if(Q.empty()) cout << -1 << "\n";
			else cout << Q.front() << "\n";
		} 
		if(cmd == "back"){
			if(Q.empty()) cout << -1 << "\n";
			else cout << Q.back() << "\n";
		} 
	}
	return 0;
} 